CONTRIBUTING.adoc

```diff
@@ -14,6 +14,8 @@ Run `./mvnw clean test` in the root directory of the project to run the tests.
 The `./mvnw` is a self-contained https://maven.apache.org/[Maven] wrapper that allows you to build the project without having Maven installed on your local machine.
 +
 You can run the tests of a specific module by using the `-f` flag like this: `./mvnw clean test -f spring-cloud-gcp-pubsub`
++
+If you are contributing to a sample application, please use `full-checkstyle` maven profile to validate code style locally: `./mvnw clean test -P full-checkstyle`.
```

Comment:
I'm not sure what is the difference then: we still require Checkstyle rules for samples and request more work to do when contribute a new one or modify existing.
So, why just don't *ignore* `ImportOrder` for samples like you did for `HideUtilityClassConstructorCheck`?

Thanks

Reply:
We have two different usecases when someone is trying to run a sample:
* people who want to try the sample apps, but don't intend to contribute any changes to the codebase. For them, checkstyle is just a distraction from what they are trying to get done -- learn how to use our integrations.
* actual contributors, who should be bound by checkstyle because our codebase should be consistent.

What this change does now is:
* guarantee that incorrect style does not make it into the codebase (because the new profile runs on Travis at install phase).
* make it possible to run full checkstyle locally (so the contributor does not get surprised by new checkstyle requirements after PR is created).
* for casual users, default to not enforcing checkstyle on sample apps .